modules/giapi/src/main/scala/giapi/client/GiapiConfig.scala

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package giapi.client
+
+import cats.Show
+
+// Produce a configuration string for Giapi.
+trait GiapiConfig[T] {
+  def configuration(t: T): String
+}
+
+object GiapiConfig {
+  def apply[A](implicit instance: GiapiConfig[A]): GiapiConfig[A] = instance
+
+  implicit class GiapiConfigOps[A](value: A) {
+    def configuration(implicit w: GiapiConfig[A]): String =
+      w.configuration(value)
+  }
+
+  def fromShow[A: Show]: GiapiConfig[A] = Show[A].show(_)
+
+  implicit val stringConfig: GiapiConfig[String] = t => t
+  implicit val intConfig: GiapiConfig[Int] = _.toString
+  implicit val doubleConfig: GiapiConfig[Double] = _.toString
```

Comment:
I wonder if this should take precision. It is not needed in GPI and probably not in GHOST but maybe they should take one. I can imagine an instrument choking when sent a string with many many decimals
We can keep it as it is for now

Reply:
More problematic (maybe) is scientific notation.

```scala
@ val a = 0.0001 
a: Double = 1.0E-4

@ a.toString 
res22: String = "1.0E-4"

@ f"$a%6.6f" 
res23: String = "0.000100"
```

Reply:
Good point

Reply:
Yes, that had occurred to me but I wasn't sure what to do with it. I'm not sure how `Show` treats `Double` but I'm curious.

Reply:
```scala
@ 0.0001 
res1: Double = 1.0E-4

@ 0.0001.show 
res2: String = "1.0E-4"
```